Dockerfile

```diff
@@ -0,0 +1,73 @@
+# build ironlib wrapper binaries
+FROM golang:1.18-alpine AS stage0
+
+WORKDIR /workspace
+
+# copy the go modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# copy ironlib go sources
+COPY . .
+
+# build helper util
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on \
+    go build -o getbiosconfig examples/biosconfig/biosconfig.go && \
+    install -m 755 -D getbiosconfig /usr/sbin/
+
+FROM almalinux:9-minimal
+
+ARG DEPDIR="dependencies"
+COPY "${DEPDIR}" dependencies
+
+# copy ironlib wrapper binaries
+COPY --from=stage0 /usr/sbin/getbiosconfig /usr/sbin/getbiosconfig
+
+# import and install tools
+RUN microdnf install -y --setopt=tsflags=nodocs crypto-policies-scripts && \
+    update-crypto-policies --set DEFAULT:SHA1 && \
+    rpm -ivh ${DEPDIR}/epel-release-latest-9.noarch.rpm
+
+## Prerequisite directories for Dell, ASRR, Supermicro
+## /lib/firmware required for Dell updates to be installed successfullly
+RUN mkdir -p /lib/firmware /opt/asrr /usr/libexec/dell_dup /opt/supermicro/sum/
+# Bootstrap Dell DSU repository
+RUN curl -O https://linux.dell.com/repo/hardware/dsu/bootstrap.cgi && bash bootstrap.cgi && rm -f bootstrap.cgi
+# Install Dell idracadm7 to enable collecting BIOS configuration and use install_weak_deps=0 to avoid pulling extra packages
+RUN microdnf install -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 \
+    srvadmin-idracadm7
+
+# install misc support packages
+RUN microdnf install -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 \
+    dmidecode     \
+    dosfstools    \
+    e2fsprogs     \
+    gdisk         \
+    gzip          \
+    hdparm        \
+    kmod          \
+    libssh2-devel \
+    lshw          \
+    mdadm         \
+    nvme-cli      \
+    pciutils      \
+    smartmontools \
+    tar           \
+    unzip         \
+    util-linux    \
+    which         \
+    microdnf clean all && \
+    ln -s /usr/bin/microdnf /usr/bin/yum # since dell dsu expects yum
+
+# Provide hook to include extra dependencies in the image
+RUN if [[ -f ${DEPDIR}/install-extra-deps.sh ]]; then cd ${DEPDIR} && bash install-extra-deps.sh; fi
+
+# Delete /tmp/* and dependencies dir as we don't need those included in the image.
+# for some reason this is not working in reducing the size of the image
```

Comment:
The image size isn't being reduced because all this is happening across multiple RUNs and they are being stored in each layer. The best solution to that problem w/o resorting to one huge RUN is to change `FROM almalinux:9-minimal` as `FROM almalinux:9-minimal AS stage1` and keep the rest of this file as is and then later do

```
FROM scratch
COPY --from=stage1 / /
```
This will give you only the files that have been kept at the end of the last command of the last stage. Effectively squashing the image layers down to just 1.

Reply:
I thought each `RUN` would create a new layer and then the last instruction would be the layer used. Apparently it doesn't work like that. Anyway, thanks for the suggestion, fixed in [dbfa026](https://github.com/metal-toolbox/ironlib/pull/84/commits/dbfa0268b18ee71b258940973c479e09eda304bf) and image is now much smaller.